### PR TITLE
Setting up root for asraa

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by asraa** [comment]
Verified the newly staged metadata (root, targets, and new delegations)
```
STAGED METADATA

Outputting metadata verification at /home/asraa/git/root-signing/ceremony/2022-05-09...

Verifying root.json...
	Contains 0/3 valid signatures
	root version 3, expires 2022/11/10

Verifying revocation.json...
	Contains 0/1 valid signatures
	revocation version 1, expires 2022/11/10

Verifying targets.json...
	Contains 0/3 valid signatures
	targets version 3, expires 2022/11/10

Verifying rekor.json...
	Contains 0/1 valid signatures
	rekor version 2, expires 2022/11/10
```

File: ceremony/2021-11-11/repository/snapshot.json [hunk 1]
```diff
@@ -2,12 +2,12 @@
 	"signatures": [
 		{
 			"keyid": "fc61191ba8a516fe386c7d6c97d918e1d241e1589729add09b122725b8c32451",
-			"sig": "30450220259b26200ac171a9ccaad7a862f477d439d9f03d5d8deae9c7f76dd6b23d5687022100ccffd7411ee5cfe2fe11841442a925b2e25eedf11c9653cdaafa36f8721ed512"
+			"sig": "3046022100d2a7627fd6287c8f8a88ae2c8e9d04ed2022a4d1628bb46a40d72d45e265c92f022100bd6f50fc8b2866ee9ff5d6d3cb27060f09adf488914f33d424a32243c04f8aa1"
```
Comment (Hayden-IO) [inline ceremony/2021-11-11/repository/snapshot.json]: Should this change? This was from the last ceremony

Comment (asraa) [reply ceremony/2021-11-11/repository/snapshot.json]: This is the final repository from the current `repository/`, I haven't staged a new snapshot.

This is so the TUF client doesn't fail on presubmit or when you run verify published ceremony data.

File: ceremony/2022-05-09/staged/root.json [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+			"sig": ""
+		},
+		{
+			"keyid": "35e596c21d78856bfab72b3af4a75dc2ba695ee26f1f0fdb4c04d4396ff5a52b",
+			"sig": ""
+		}
+	],
+	"signed": {
+		"_type": "root",
+		"consistent_snapshot": false,
```
Comment (Hayden-IO) [inline ceremony/2022-05-09/staged/root.json]: Shouldnt this be true?

**Comment by bobcallaway** [comment]
```sh
bcallaway-macbookpro:root-signing bcallaway$ export REPO=$(pwd)/ceremony/2022-05-09
bcallaway-macbookpro:root-signing bcallaway$ ./scripts/verify.sh 206
+ '[' -z bobcallaway ']'
+ '[' -z /Users/bcallaway/git/sigstore/root-signing/ceremony/2022-05-09 ']'
+ '[' -z v3-staging ']'
+ echo 'Using branch v3-staging'
Using branch v3-staging
+ git remote rm upstream
+ git remote add upstream git@github.com:sigstore/root-signing.git
+ git remote rm origin
+ git remote add origin git@github.com:bobcallaway/root-signing.git
+ git remote -v
origin	git@github.com:bobcallaway/root-signing.git (fetch)
origin	git@github.com:bobcallaway/root-signing.git (push)
upstream	git@github.com:sigstore/root-signing.git (fetch)
upstream	git@github.com:sigstore/root-signing.git (push)
+ git checkout v3-staging
Switched to branch 'v3-staging'
+ git pull upstream v3-staging
From github.com:sigstore/root-signing
 * branch            v3-staging -> FETCH_HEAD
 * [new branch]      v3-staging -> upstream/v3-staging
Already up to date.
+ git status
On branch v3-staging
nothing to commit, working tree clean
+ git remote -v
origin	git@github.com:bobcallaway/root-signing.git (fetch)
origin	git@github.com:bobcallaway/root-signing.git (push)
upstream	git@github.com:sigstore/root-signing.git (fetch)
upstream	git@github.com:sigstore/root-signing.git (push)
+ go build -o verify ./cmd/verify
+ '[' -f piv-attestation-ca.pem ']'
+ [[ ! -z 206 ]]
+ echo 'Pull Request: 206'
Pull Request: 206
+ git branch -D VERIFY
Deleted branch VERIFY (was b0e55ba).
+ git fetch upstream pull/206/head:VERIFY
From github.com:sigstore/root-signing
 * [new ref]         refs/pull/206/head -> VERIFY
+ git checkout VERIFY
Switched to branch 'VERIFY'
+ ./verify keys --root piv-attestation-ca.pem --key-directory /Users/bcallaway/git/sigstore/root-signing/ceremony/2022-05-09/keys

Outputting key verification and OpenSSL commands...

VERIFIED KEY WITH SERIAL NUMBER 13078778
	TUF key id: 2f64fb5eac0cf94dd39bb45308b98920055e9a0d8e012a7220787834c60aef97

VERIFIED KEY WITH SERIAL NUMBER 14470876
	TUF key id: eaf22372f417dd618a46f6c627dbc276e9fd30a004fc94f9be946e73f8bd090b

VERIFIED KEY WITH SERIAL NUMBER 15938765
	TUF key id: f40f32044071a9365505da3d1e3be6561f6f22d0e60cf51df783999f6c3429cb

VERIFIED KEY WITH SERIAL NUMBER 15938791
	TUF key id: f505595165a177a41750a8e864ed1719b1edfccd5a426fd2c0ffda33ce7ff209

VERIFIED KEY WITH SERIAL NUMBER 18158855
	TUF key id: 35e596c21d78856bfab72b3af4a75dc2ba695ee26f1f0fdb4c04d4396ff5a52b

# To manually verify the chain for any key ID

	export SERIAL_NUMBER=${SERIAL_NUMBER}
	openssl verify -verbose -x509_strict -CAfile <(cat piv-attestation-ca.pem ceremony/2022-05-09/keys/${SERIAL_NUMBER}/${SERIAL_NUMBER}_device_cert.pem) ceremony/2022-05-09/keys/${SERIAL_NUMBER}/${SERIAL_NUMBER}_key_cert.pem

# Manually extract the public key for any key ID and match with published

	export SERIAL_NUMBER=${SERIAL_NUMBER}
	openssl x509 -in ceremony/2022-05-09/keys/${SERIAL_NUMBER}/${SERIAL_NUMBER}_pubkey.pem -pubkey -noout
	cat ceremony/2022-05-09/keys/${SERIAL_NUMBER}/${SERIAL_NUMBER}_key_cert.pem
+ '[' -f /Users/bcallaway/git/sigstore/root-signing/ceremony/2022-05-09/repository/1.root.json ']'
+ ./verify repository --repository /Users/bcallaway/git/sigstore/root-signing/ceremony/2022-05-09 --staged
STAGED METADATA

Outputting metadata verification at /Users/bcallaway/git/sigstore/root-signing/ceremony/2022-05-09...

Verifying root.json...
	Contains 0/3 valid signatures
	root version 3, expires 2022/11/10

Verifying revocation.json...
	Contains 0/1 valid signatures
	revocation version 1, expires 2022/11/10

Verifying rekor.json...
	Contains 0/1 valid signatures
	rekor version 2, expires 2022/11/10

Verifying targets.json...
	Contains 0/3 valid signatures
	targets version 3, expires 2022/11/10
+ '[' -f /Users/bcallaway/git/sigstore/root-signing/ceremony/2022-05-09/repository/1.root.json ']'
+ ./verify repository --repository /Users/bcallaway/git/sigstore/root-signing/ceremony/2022-05-09 --root /Users/bcallaway/git/sigstore/root-signing/ceremony/2022-05-09/repository/1.root.json

VERIFYING TUF CLIENT UPDATE

Client successfully initialized, updating and downloading targets...
Client updated to...
	root.json version 2, expires 2022/05/11
	timestamp.json version 28, expires 2022/05/22
	snapshot.json version 28, expires 2022/05/22
	targets.json version 2, expires 2022/05/11

Retrieved target artifact.pub...
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEhyQCx0E9wQWSFI9ULGwy3BuRklnt
IqozONbbdbqz11hlRJy9c7SG+hdcFl9jE9uE/dwtuwU2MqU9T/cN0YkWww==
-----END PUBLIC KEY-----

Retrieved target ctfe.pub...
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEbfwR+RJudXscgRBRpKX1XFDy3Pyu
dDxz/SfnRi1fT8ekpfBd2O1uoz7jr3Z8nKzxA69EUQ+eFCFI3zeubPWU7w==
-----END PUBLIC KEY-----

Retrieved target fulcio.crt.pem...
-----BEGIN CERTIFICATE-----
MIIB+DCCAX6gAwIBAgITNVkDZoCiofPDsy7dfm6geLbuhzAKBggqhkjOPQQDAzAq
MRUwEwYDVQQKEwxzaWdzdG9yZS5kZXYxETAPBgNVBAMTCHNpZ3N0b3JlMB4XDTIx
MDMwNzAzMjAyOVoXDTMxMDIyMzAzMjAyOVowKjEVMBMGA1UEChMMc2lnc3RvcmUu
ZGV2MREwDwYDVQQDEwhzaWdzdG9yZTB2MBAGByqGSM49AgEGBSuBBAAiA2IABLSy
A7Ii5k+pNO8ZEWY0ylemWDowOkNa3kL+GZE5Z5GWehL9/A9bRNA3RbrsZ5i0Jcas
taRL7Sp5fp/jD5dxqc/UdTVnlvS16an+2Yfswe/QuLolRUCrcOE2+2iA5+tzd6Nm
MGQwDgYDVR0PAQH/BAQDAgEGMBIGA1UdEwEB/wQIMAYBAf8CAQEwHQYDVR0OBBYE
FMjFHQBBmiQpMlEk6w2uSu1KBtPsMB8GA1UdIwQYMBaAFMjFHQBBmiQpMlEk6w2u
Su1KBtPsMAoGCCqGSM49BAMDA2gAMGUCMH8liWJfMui6vXXBhjDgY4MwslmN/TJx
Ve/83WrFomwmNf056y1X48F9c4m3a3ozXAIxAKjRay5/aj/jsKKGIkmQatjI8uup
Hr/+CxFvaJWmpYqNkLDGRU+9orzh5hI2RrcuaQ==
-----END CERTIFICATE-----

Retrieved target fulcio_v1.crt.pem...
-----BEGIN CERTIFICATE-----
MIIB9zCCAXygAwIBAgIUALZNAPFdxHPwjeDloDwyYChAO/4wCgYIKoZIzj0EAwMw
KjEVMBMGA1UEChMMc2lnc3RvcmUuZGV2MREwDwYDVQQDEwhzaWdzdG9yZTAeFw0y
MTEwMDcxMzU2NTlaFw0zMTEwMDUxMzU2NThaMCoxFTATBgNVBAoTDHNpZ3N0b3Jl
LmRldjERMA8GA1UEAxMIc2lnc3RvcmUwdjAQBgcqhkjOPQIBBgUrgQQAIgNiAAT7
XeFT4rb3PQGwS4IajtLk3/OlnpgangaBclYpsYBr5i+4ynB07ceb3LP0OIOZdxex
X69c5iVuyJRQ+Hz05yi+UF3uBWAlHpiS5sh0+H2GHE7SXrk1EC5m1Tr19L9gg92j
YzBhMA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBRY
wB5fkUWlZql6zJChkyLQKsXF+jAfBgNVHSMEGDAWgBRYwB5fkUWlZql6zJChkyLQ
KsXF+jAKBggqhkjOPQQDAwNpADBmAjEAj1nHeXZp+13NWBNa+EDsDP8G1WWg1tCM
WP/WHPqpaVo0jhsweNFZgSs0eE7wYI4qAjEA2WB9ot98sIkoF3vZYdd3/VtWB5b9
TNMea7Ix/stJ5TfcLLeABLE4BNJOsQ4vnBHJ
-----END CERTIFICATE-----

Retrieved target rekor.pub...
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE2G2Y+2tabdTV5BcGiBIx0a9fAFwr
kBbmLSGtks4L3qX6yYY0zufBnhC8Ur/iy55GhWP/9A/bY2LhC30M9+RYtw==
-----END PUBLIC KEY-----
bcallaway-macbookpro:root-signing bcallaway$ 
```

File: ceremony/2022-05-09/staged/root.json (hunk 2 above)
Comment (asraa) [reply ceremony/2022-05-09/staged/root.json]: Yeah, but I'm not sure I can debug why this line didn't hold
https://github.com/sigstore/root-signing/blob/c6a6af4c78f22ce2641c30d5605d0473d68d589d/cmd/tuf/app/init.go#L193

I'd have to dig in go-tuf code

File: ceremony/2021-11-11/repository/snapshot.json (hunk 1 above)
Comment (Hayden-IO) [reply ceremony/2021-11-11/repository/snapshot.json]: Isn't this an issue that we're changing what was done at the previous ceremony though? For auditability

Comment (asraa) [reply ceremony/2021-11-11/repository/snapshot.json]: I can restore these old ceremony, although the history still maintained, I only needed it for the current one.

File: ceremony/2022-05-09/staged/root.json (hunk 2 above)
Comment (asraa) [reply ceremony/2022-05-09/staged/root.json]: per offline, changing to true.

**Comment by dlorenc** [comment]
I got the same as @bobcallaway above

**Comment by dlorenc** [comment]
LGTM, I see version 3 and the 6 month date.

**Comment by dlorenc** [comment]
Still looks good after the force push.

I reviewed the delta as well.

**Review by dlorenc — APPROVED** [review]

**Comment by Hayden-IO** [comment]
Do the targets become versioned from the consistent snapshot later on?

**Comment by asraa** [comment]
> Do the targets become versioned from the consistent snapshot later on?

 Yes, at the publish step in go-tuf code is the only time files are written to the final store and consistent snapshot's taken into account
 
 https://github.com/theupdateframework/go-tuf/blob/4bf58eb096f99647e7fd30447396c7a57202982f/repo.go#L1495

**Comment by Hayden-IO** [comment]
One last question, `3.root.json` is on commit?

**Review by Hayden-IO — APPROVED** [review]

**Comment by asraa** [comment]
>  One last question, 3.root.json is on commit?



Yes, same code that's calling the ConsistentSnapshot versioned commit.

**Pull request merged**